In [774]:
#### Contributor : Shree Priya

In [775]:
#Load the required libraries
import numpy as np
import pandas as pd
import scipy.stats as stats
import statsmodels.formula.api as smf
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings('ignore')

In [776]:
# Read the data
data = pd.read_csv("progresa-sample.csv.bz2")
data.head()

year  sex  indig  dist_sec   sc  grc  fam_n   min_dist   dist_cap   poor  \
0    97  0.0    0.0     4.473  1.0  7.0      7  21.168384  21.168384  pobre   
1    98  0.0    0.0     4.473  1.0  8.0      7  21.168384  21.168384  pobre   
2    97  1.0    0.0     4.473  1.0  6.0      7  21.168384  21.168384  pobre   
3    98  1.0    0.0     4.473  1.0  7.0      7  21.168384  21.168384  pobre   
4    97  0.0    0.0     4.473  1.0  2.0      7  21.168384  21.168384  pobre   

   ... hohedu  hohwag  welfare_index  hohsex  hohage  age  village  folnum  \
0  ...      6     0.0          583.0     1.0    35.0   13      163       1   
1  ...      6     0.0          583.0     1.0    35.0   14      163       1   
2  ...      6     0.0          583.0     1.0    35.0   12      163       2   
3  ...      6     0.0          583.0     1.0    35.0   13      163       2   
4  ...      6     0.0          583.0     1.0    35.0    8      163       3   

   grc97  sc97  
0      7   1.0  
1      7   1.0  
2      6   1.0  
3      6   1.0  
4      2   1.0  

[5 rows x 21 columns]

In [777]:
# Dimensions of data
data.shape

(77250, 21)

In [778]:
# Attributes in data
data.columns

Index(['year', 'sex', 'indig', 'dist_sec', 'sc', 'grc', 'fam_n', 'min_dist',
       'dist_cap', 'poor', 'progresa', 'hohedu', 'hohwag', 'welfare_index',
       'hohsex', 'hohage', 'age', 'village', 'folnum', 'grc97', 'sc97'],
      dtype='object')

### Question 1.1 

In [779]:
# Explore values of progres
data.progresa.unique()

array(['0', 'basal'], dtype=object)

In [780]:
# Changing coding scheme of progresa
data['progresa'] = np.where(data['progresa'] == 'basal' ,1.0,0.0)


In [781]:
# Check values in progresa attribute
data.progresa.unique()

array([0., 1.])

In [782]:
# Reporting summary statistics
data_describe = pd.DataFrame(data[['sex', 'indig', 'dist_sec', 'sc', 'grc', 'fam_n', 'min_dist',
       'dist_cap', 'poor', 'progresa', 'hohedu', 'hohwag', 'welfare_index',
       'hohsex', 'hohage', 'age', 'grc97', 'sc97']].describe())

In [783]:
data_describe = data_describe.reset_index()

In [784]:
data_stats = data_describe.pivot_table(columns=['index'])

# Presenting summary statistics in table form
data_stats

index                 25%         50%         75%    count           max  \
age              9.000000   11.000000   14.000000  77250.0     17.000000   
dist_cap        92.327050  132.001494  184.445225  77250.0    359.774457   
dist_sec         0.574000    2.279000    3.582000  77250.0     14.879000   
fam_n            6.000000    7.000000    9.000000  77250.0     24.000000   
grc              2.000000    4.000000    6.000000  70701.0     14.000000   
grc97            1.000000    4.000000    6.000000  77250.0     14.000000   
hohage          36.000000   43.000000   51.000000  77240.0     98.000000   
hohedu           0.000000    2.000000    4.000000  77250.0     20.000000   
hohsex           1.000000    1.000000    1.000000  77230.0      1.000000   
hohwag         120.000000  500.000000  750.000000  77250.0  14000.000000   
indig            0.000000    0.000000    1.000000  76950.0      1.000000   
min_dist        70.518238  111.228612  138.446009  77250.0    170.457647   
progresa         0.000000    1.000000    1.000000  77250.0      1.000000   
sc               1.000000    1.000000    1.000000  68797.0      1.000000   
sc97             1.000000    1.000000    1.000000  73378.0      1.000000   
sex              0.000000    1.000000    1.000000  77226.0      1.000000   
welfare_index  597.000000  685.000000  770.000000  77040.0   1294.000000   

index                mean         min         std  
age             11.366460    6.000000    3.167744  
dist_cap       147.674452    9.465392   76.063134  
dist_sec         2.418910    0.000000    2.234109  
fam_n            7.215715    1.000000    2.352900  
grc              3.963537    0.000000    2.499063  
grc97            3.705372    0.000000    2.572387  
hohage          44.436717   15.000000   11.620372  
hohedu           2.768104    0.000000    2.656106  
hohsex           0.925185    0.000000    0.263095  
hohwag         586.985312    0.000000  788.133664  
indig            0.298324    0.000000    0.457525  
min_dist       103.447520    9.465392   42.089441  
progresa         0.615663    0.000000    0.486441  
sc               0.819818    0.000000    0.384342  
sc97             0.813922    0.000000    0.389172  
sex              0.512211    0.000000    0.499854  
welfare_index  690.346564  180.000000  139.491130

### Question 1.2.1 and 1.2.2

In [785]:
# Filtering data for poor in the year 1997
data_new = data[((data.year == 97) & (data.poor == 'pobre'))]
data_new.head()

year  sex  indig  dist_sec   sc  grc  fam_n    min_dist    dist_cap   poor  \
0    97  0.0    0.0     4.473  1.0  7.0      7   21.168384   21.168384  pobre   
2    97  1.0    0.0     4.473  1.0  6.0      7   21.168384   21.168384  pobre   
4    97  0.0    0.0     4.473  1.0  2.0      7   21.168384   21.168384  pobre   
6    97  0.0    0.0     3.154  0.0  6.0      6  127.114780  154.196003  pobre   
8    97  1.0    0.0     3.373  1.0  2.0      5   85.300272  105.878669  pobre   

   ...  hohedu  hohwag  welfare_index  hohsex  hohage  age  village  folnum  \
0  ...       6     0.0      583.00000     1.0    35.0   13      163       1   
2  ...       6     0.0      583.00000     1.0    35.0   12      163       2   
4  ...       6     0.0      583.00000     1.0    35.0    8      163       3   
6  ...       4     0.0      684.00000     1.0    85.0   14      271       4   
8  ...       6   875.0      742.14001     1.0    26.0    9      263       5   

   grc97  sc97  
0      7   1.0  
2      6   1.0  
4      2   1.0  
6      6   0.0  
8      2   1.0  

[5 rows x 21 columns]

In [786]:
# Creating a new column 'treatment' based on progresa
data_new['treatment'] = np.where(data_new['progresa'] == 0.0, 'Control', 'Treatment')

In [787]:
# Seggregating Control group 
control = data_new[data_new['treatment'] == 'Control']
treatment = data_new[data_new['treatment'] == 'Treatment']

# Selecting rows where poor=1 and the year=97, and then grouping by 'progresa' column
ttest = data_new.groupby('progresa').mean()
ttest.drop(ttest.columns[[0,9,16,17]], axis =1,inplace=True)

ttest = ttest.transpose()
# Swap columns to match the structure
ttest = ttest[[1,0]]

# Resetting Index
ttest.reset_index(level=0, inplace=True)

ttest.rename(columns={'index' : 'Variable', 0.0: 'Avg(Control)', 1.0: 'Avg(Treatment villages)'}, inplace=True)

# List of all Variables
var = list(ttest['Variable'])
# Calculating T test for the Treatment and Control groups
tt = list(stats.ttest_ind(treatment[var], control[var], nan_policy='omit'))

# Adding the remaining two columns.
ttest['Difference'] = tt[0]
ttest['p-value'] = tt[1]

ttest

progresa       Variable  Avg(Treatment villages)  Avg(Control)  Difference  \
0                   sex                 0.519317      0.505052    2.506686   
1                 indig                 0.325986      0.332207   -1.161714   
2              dist_sec                 2.453122      2.507662   -2.100433   
3                    sc                 0.822697      0.815186    1.668745   
4                   grc                 3.531599      3.543050   -0.400196   
5                 fam_n                 7.281327      7.302469   -0.794167   
6              min_dist               107.152915    103.237854    8.206584   
7              dist_cap               150.829074    153.769730   -3.339081   
8                hohwag               544.339544    573.163558   -3.594588   
9         welfare_index               655.428377    659.579100   -3.188594   
10               hohsex                 0.924656      0.922947    0.566312   
11               hohage                43.648828     44.276918   -4.775962   
12                  age                10.716991     10.742023   -0.708630   
13              village               233.795025    286.363155  -30.802415   
14                 sc97                 0.822697      0.815186    1.668745   

progresa        p-value  
0          1.219172e-02  
1          2.453603e-01  
2          3.569843e-02  
3          9.517806e-02  
4          6.890151e-01  
5          4.271039e-01  
6          2.358312e-16  
7          8.415005e-04  
8          3.253835e-04  
9          1.431016e-03  
10         5.711858e-01  
11         1.796243e-06  
12         4.785594e-01  
13        2.114020e-205  
14         9.517806e-02

### Question 1.2.3
Here I have applied double-sided t-test to find the significant difference between control and treatmenrt groups.
According to the study, the treatment and control groups were selected randomly.

The following variables has the p-value < 0.5 : 
Sex, min_dist(min distance to an urban center), dist_cap (min distance to the capital),dsit_sec(nearest distance to a secondary school), hohedu (years of schooling of head of household), hohwag (monthly wages of head of household), welfare_index, hohage((age of head of household)

For these variables there is statistically significant differences between treatment and control villages as baseline.


### Question 1.2.4
Statistically significant differences between treatment and control villages matters because if there are differences at baseline, it tells us that the distribution of Treatment and Control group is not completely random. It shows that these groups are statistically different from each other.
In order to perform accurate casual analysis, it's required to have a  indistinguishable baseline.

### Question 1.2.5
As some of the variables in Treatment and Control group have statistically significant differences,it might impact the result. For example,
if we want to assess the impact of progresa in the Treatment group, we won't be able to justify that the student enrollment improved solely becuase of progresa program.
Other factors like sex, minimum distance to the capital, might also influence the result as they are statistically different from each other at the baseline.

In [788]:
# Question 2 Measuring Impact
# Filtering the data
data = data[data['poor'] == 'pobre']
data = data.dropna()

In [789]:
# Creating a 'before' column for evaluating impact before progresa
data['before'] = data.year == 97

### Question 2.1 Before-after estimator

In [790]:
# Question 2.1.1
# Comparing the average schooling rates
data[data.progresa == 1.0].groupby('before').sc.mean()

before
False    0.849257
True     0.822697
Name: sc, dtype: float64

In [791]:
### We found that on an average, there's an increase in enrollment rate by 0.0265 after the progresa

In [792]:
# Linear Regression model
m = smf.ols(formula = 'sc ~ before', data=data[data.progresa == 1.0])
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     45.02
Date:                Wed, 05 Feb 2020   Prob (F-statistic):           1.98e-11
Time:                        11:32:49   Log-Likelihood:                -15111.
No. Observations:               35355   AIC:                         3.023e+04
Df Residuals:                   35353   BIC:                         3.024e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          0.8493      0.003    292.114      0.000       0.844       0.855
before[T.True]    -0.0266      0.004     -6.709      0.000      -0.034      -0.019
==============================================================================
Omnibus:                    10153.494   Durbin-Watson:                   1.388
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            21422.127
Skew:                          -1.801   Prob(JB):                         0.00
Kurtosis:                       4.252   Cond. No.                         2.72
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### The coefficient of estimator using linear regression suggest that there is increase in enrollment rate by 0.0266 after the progresa

In [793]:
# Multiple Regression model
m = smf.ols(formula = 'sc ~ before + dist_sec + sex + min_dist + dist_cap + hohedu', data=data[data.progresa == 1.0])
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.022
Model:                            OLS   Adj. R-squared:                  0.022
Method:                 Least Squares   F-statistic:                     134.5
Date:                Wed, 05 Feb 2020   Prob (F-statistic):          4.77e-169
Time:                        11:32:49   Log-Likelihood:                -14734.
No. Observations:               35355   AIC:                         2.948e+04
Df Residuals:                   35348   BIC:                         2.954e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          0.7401      0.007    103.416      0.000       0.726       0.754
before[T.True]    -0.0255      0.004     -6.499      0.000      -0.033      -0.018
dist_sec          -0.0068      0.001     -7.042      0.000      -0.009      -0.005
sex                0.0273      0.004      6.981      0.000       0.020       0.035
min_dist           0.0003   6.47e-05      4.537      0.000       0.000       0.000
dist_cap           0.0003   3.53e-05      7.259      0.000       0.000       0.000
hohedu             0.0146      0.001     18.704      0.000       0.013       0.016
==============================================================================
Omnibus:                     9695.850   Durbin-Watson:                   1.408
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            19839.800
Skew:                          -1.739   Prob(JB):                         0.00
Kurtosis:                       4.174   Cond. No.                         768.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### The coefficient of estimator using multiple regression suggest that there is increase in enrollment rate by 0.0255 after the progresa

#### The values of estimators are 0.0265, 0.0266, and 0.0255. There is not much difference in the values of estimators.

#### The p value of the parameter obtained is not greater than 0.05, hence these estimates are statistically significant.
#### The coefficient of parameter 'before' the progresa is negative, which shows that there was a positive impact 'after' the progresa on the student enrollment.


### Question 2.2

In [794]:
data[data['before']==False].groupby('progresa').sc.mean()

progresa
0.0    0.810923
1.0    0.849257
Name: sc, dtype: float64

#### The differenece between average enrollment rate among poor households in the treatment villages and the average enrollment rate among poor households in the control villages is 0.03826

In [795]:
# Linear Regression model
m = smf.ols(formula = 'sc ~ progresa', data=data)
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     44.28
Date:                Wed, 05 Feb 2020   Prob (F-statistic):           2.87e-11
Time:                        11:32:49   Log-Likelihood:                -25434.
No. Observations:               56893   AIC:                         5.087e+04
Df Residuals:                   56891   BIC:                         5.089e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8132      0.003    315.401      0.000       0.808       0.818
progresa       0.0218      0.003      6.655      0.000       0.015       0.028
==============================================================================
Omnibus:                    15154.174   Durbin-Watson:                   1.388
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            30457.736
Skew:                          -1.724   Prob(JB):                         0.00
Kurtosis:                       3.978   Cond. No.                         3.01
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### The coefficient of estimator using linear regression suggest that there is increase in average enrollment rate among poor households in the treatment  villages by 0.0218  because of progresa

In [796]:
# Multiple regression model
m = smf.ols(formula = 'sc ~ progresa + dist_sec + sex + min_dist + dist_cap + hohedu', data=data)
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.022
Model:                            OLS   Adj. R-squared:                  0.022
Method:                 Least Squares   F-statistic:                     210.1
Date:                Wed, 05 Feb 2020   Prob (F-statistic):          3.43e-266
Time:                        11:32:49   Log-Likelihood:                -24832.
No. Observations:               56893   AIC:                         4.968e+04
Df Residuals:                   56886   BIC:                         4.974e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.7041      0.006    126.469      0.000       0.693       0.715
progresa       0.0193      0.003      5.950      0.000       0.013       0.026
dist_sec      -0.0066      0.001     -8.970      0.000      -0.008      -0.005
sex            0.0283      0.003      9.018      0.000       0.022       0.034
min_dist       0.0004   5.17e-05      7.263      0.000       0.000       0.000
dist_cap       0.0002   2.78e-05      7.342      0.000       0.000       0.000
hohedu         0.0150      0.001     23.815      0.000       0.014       0.016
==============================================================================
Omnibus:                    14470.751   Durbin-Watson:                   1.408
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            28273.425
Skew:                          -1.665   Prob(JB):                         0.00
Kurtosis:                       3.915   Cond. No.                         753.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

####  The coefficient of estimator using multiple regression suggest that there is increase in average enrollment rate among poor households in the treatment villages by 0.0193  because of progresa
#### There are other variables with significant t-value like hohedu (years of schooling of head of household) which has an impact an increase in average enrollment rate among poor households in the treatment villages 

#### The values of estimators are 0.03826, 0.0218, and 0.0193. The coefficient of estimators have slighly decreased in linear and multiple regression.

#### The p value of the parameter obtained is not greater than 0.05, hence these estimates are statistically significant.
#### The coefficient associated with progresa is positive, which shows that there was overall a positive impact of progresa on the student enrollment.

#### In mutiple regression model, there are other parameters also which has statistically significant effect on the average enrollment rate. There are other variables with significant t-value like hohedu (years of schooling of head of household) which has an impact an increase in average enrollment rate among poor households in the treatment villages


### Question 2.3

In [797]:
# Create a new column to capture after effect
data['after'] = data.year == 98

In [798]:
data.groupby(['progresa', 'after'], as_index = False).sc.mean()

progresa  after        sc
0       0.0  False  0.815066
1       0.0   True  0.810923
2       1.0  False  0.822697
3       1.0   True  0.849257

In [799]:
(0.849257 - 0.822697) - (0.810923 - 0.815066)

0.030703000000000036

#### The Difference between average enrollment rate among poor households in the treatment villages and the average enrollment rate among poor households in the control villages, both 1997 and 1998  is 0.0307
#### From the table above, we observed that the impact of progresa program has a positive impact on student enrollment.

In [800]:
# Linear Regression
m = smf.ols(formula = 'sc ~ progresa * after', data=data)
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.002
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     29.42
Date:                Wed, 05 Feb 2020   Prob (F-statistic):           5.32e-19
Time:                        11:32:49   Log-Likelihood:                -25412.
No. Observations:               56893   AIC:                         5.083e+04
Df Residuals:                   56889   BIC:                         5.087e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Intercept                  0.8151      0.004    232.782      0.000       0.808       0.822
after[T.True]             -0.0041      0.005     -0.801      0.423      -0.014       0.006
progresa                   0.0076      0.004      1.717      0.086      -0.001       0.016
progresa:after[T.True]     0.0307      0.007      4.680      0.000       0.018       0.044
==============================================================================
Omnibus:                    15130.097   Durbin-Watson:                   1.392
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            30379.093
Skew:                          -1.722   Prob(JB):                         0.00
Kurtosis:                       3.976   Cond. No.                         7.63
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### The coefficient of estimator using linear regression suggest that there is increase in average enrollment rate among poor households in the treatment villages and the average enrollment rate among poor households in the control villages, both 1997 and 1998 by 0.0307

In [801]:
# Multiple Regression
m = smf.ols(formula = 'sc ~ progresa * after + dist_sec + welfare_index + sex + min_dist + dist_cap + hohedu', data=data)
r = m.fit()
r.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.025
Model:                            OLS   Adj. R-squared:                  0.025
Method:                 Least Squares   F-statistic:                     162.5
Date:                Wed, 05 Feb 2020   Prob (F-statistic):          1.98e-305
Time:                        11:32:50   Log-Likelihood:                -24734.
No. Observations:               56893   AIC:                         4.949e+04
Df Residuals:                   56883   BIC:                         4.958e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Intercept                  0.8206      0.011     75.050      0.000       0.799       0.842
after[T.True]             -0.0058      0.005     -1.131      0.258      -0.016       0.004
progresa                   0.0043      0.004      0.981      0.327      -0.004       0.013
progresa:after[T.True]     0.0306      0.006      4.714      0.000       0.018       0.043
dist_sec                  -0.0076      0.001    -10.213      0.000      -0.009      -0.006
welfare_index             -0.0002   1.38e-05    -12.491      0.000      -0.000      -0.000
sex                        0.0287      0.003      9.143      0.000       0.023       0.035
min_dist                   0.0004   5.17e-05      7.338      0.000       0.000       0.000
dist_cap                   0.0002   2.77e-05      7.503      0.000       0.000       0.000
hohedu                     0.0154      0.001     24.548      0.000       0.014       0.017
==============================================================================
Omnibus:                    14379.223   Durbin-Watson:                   1.413
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            27986.700
Skew:                          -1.657   Prob(JB):                         0.00
Kurtosis:                       3.910   Cond. No.                     4.99e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.99e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

#### The coefficient of estimator using linear regression suggest that there is increase in average enrollment rate among poor households in the treatment villages and the average enrollment rate among poor households in the control villages, both 1997 and 1998 by 0.0306

#### As observed the value of estimators are 0.0307,0.0307 and 0.0306. There is not much difference in the estimators.
####  All of them shows a positive relationship between progresa and average student enrollment.

####  The p value of the parameter obtained is not greater than 0.05, hence these estimates are statistically significant. In mutiple regression model, there are other parameters also which has statistically significant effect on the average enrollment rate.

####  In mutiple regression model, there are other parameters also which has statistically significant effect on the average enrollment rate. There are other variables with significant t-value like hohedu (years of schooling of head of household) which has an impact an increase in average enrollment rate among poor households in the treatment villages

### Question 2.4.1

The counterfactual assumptions behind all three models are -

1. Before-after Estimator - The counter factual assumption underlying this model is that there is no unobserved trend before and after the progresa program.
And the average enrollement rate of the treatment villages would have been same if progresa program wouldn't
have occurred. 

2. Cross-sectional estimator - The counter factual assumption is that average difference in outcomes between Treated and Control group is solely due to the treatment and no other factor.

3. Differences-in-differences estimator - The counter factual assumption underlying this model is that time trends for the Treated and Control groups would have been same in the abscence of progresa.And the treatment and control group must follow the same trend in the year 1997.

The counter factual assumption of Before-after Estimator is less plausible as there can be other trends or confounding variables which can increase the average enrollment in schools over a period of time and the average enrollement rate of the treatment villages could actually change even if progresa program wouldn't have occurred. Hence, we need a control group which is not affected by progresa to compare the effect of progresa.

This comparison helps us to eliminate the confounding effect after the treatment and arrive at a real casual impact because of progresa.

### Question 2.4.2

From the analysis I observed the following -

1. I found some statistically significant differences between treatment and control groups. Even though the divison is random according to the study, it doesn't meet the requirements of randomness completely.
Hence, there is some flaw in the baseline and it makes our further analysis less reliable.

2. According to Before-after estimator,Cross-sectional and Differences-in-differences estimator, we observed a positive impact of progresa on the average enrollment in school. We saw the highest impact of progresa in Difference in Difference estimator and it is the most reliable method to estimate the effect of progresa.
This is because Difference in Difference relaxes the underlying assumption of Before-after and cross-sectional estimator and accurately calculates the average enrollment rate among poor households in the treatment villages and the average enrollment rate among poor households in the control villages, both 1997 and 1998.
